### PR TITLE
[62] Create API endpoint for hacknight details

### PR DESCRIPTION
**Pull request by w1stler**
To test, use Postman:
GET ```localhost:5000/hacknights/1/details/```

### CODE REVIEW

**Comment by w1stler** [comment]
+ frontend test fixes for master branch

File: backend/app.py [hunk 1]
```diff
@@ -40,3 +40,4 @@ def initialize_extensions(app):
 api.add_resource(ParticipantsList, "/participants/")
 api.add_resource(ParticipantDetails, "/participants/<int:id>/")
 api.add_resource(HacknightList, "/hacknights/")
+api.add_resource(HacknightDetails, "/hacknights/<int:id>/details/")
```
Comment (magul) [inline backend/app.py]: also it shouldn't have `/details/` at the end

File: backend/resources/hacknight.py [hunk 2]
```diff
@@ -44,3 +44,12 @@ def post(self):
 
         return {'message': 'Hacknight created successfully.',
                 "hacknight": data}, HTTPStatus.CREATED
+
+
+class HacknightDetails(Resource):
+    def get(self, id):
+        hacknight_schema = HacknightSchema(only=('date', 'id', 'participants'))
```
Comment (magul) [inline backend/resources/hacknight.py]: why do we limit fields using `only` here?

Comment (w1stler) [reply backend/resources/hacknight.py]: This was intentional, as the ticket stated this (only these fields to be returned). Currently we don't have others, so could be removed. :-)

Comment (magul) [reply backend/resources/hacknight.py]: I believe it should be removed. I don't even see the word `only` in issue description so would assume that was to point out all fields in that model.

File: backend/tests/conftest.py [hunk 3]
```diff
@@ -132,3 +132,12 @@ def add_hacknights(app, _db):
     for _ in range(10):
         HacknightFactory.create()
     _db.session.commit()
+
+
+@pytest.fixture
+def add_participants_to_hacknight(app, _db):
```
Comment (magul) [inline backend/tests/conftest.py]: We can simplify it:
```python
@pytest.fixture
def add_participants_to_hacknight(app, _db):
    HacknightFactory(participants=ParticipantFactory.create_batch(10))
    _db.session.commit()
```

**Pull request merged**